js/core.js

```diff
@@ -666,7 +666,46 @@ var ERMrest = (function (module) {
             }
 
             return result;
-        }
+        },
+
+        // figure out if Table is pure and binary association table.
+        // binary: Has 2 outbound foreign keys and there is a composite key constraint combining all the columns from both foreign keys.
+        // pure: There are no other columns apart from the foreign key columns and columns that are part of key and have serial4 type.
+        _isPureBinaryAssociation: function () {
+            if(this._isPureBinaryAssociation_cached === undefined) {
+                this._isPureBinaryAssociation_cached = this._computePureBinaryAssociation();
+            }
+            return this._isPureBinaryAssociation_cached;
+        },
+
+        _computePureBinaryAssociation: function () {
+            if (this.referredBy.length() > 0 || this.foreignKeys.length() != 2) {
+                return false; // not binary
+            }
+
+            var fkColset = new ColSet(this.foreignKeys.colsets().reduce(function(res, colset){
+                return res.concat(colset.columns);
+            }, [])); // set of foreignkey columns
+
+            var tempKeys = this.keys.all().filter(function(key) {
+                var keyCols = key.colset.columns;
+                return !(keyCols.length == 1 && keyCols[0].type.name == "serial4"  && !(keyCols[0] in fkColset.columns));
+            }); // the key that should contain foreign key columns.
+
+            if (tempKeys.length != 1 || !fkColset._equals(tempKeys[0].colset)) {
+                return false; // not binary
+            }
+
+            // check for purity:
+            for (var i = 0, col; i < this.columns.length(); i++) {
```

Comment:
Ah, I understand now.

As long as the pair of foreign keys forms the key of the table, it means each row associates one foreign key combination. That's all I have meant by "pure" association. The kind of association table that is really impure is one that includes other material into that key combination. Then, more than one row could associate the same pair of foreign keys but with different modifying material, e.g. association modifiers. This means you can get duplicative cross-products of associated entities if you treat it as a simple association and ignore the modifier values.

So, we don't need to exclude other non-key material, but I guess you can if you want.  If you do, it should be through another colset operation rather than this loop. Create a colset of all columns, filter out serial columns, and check that the remaining colset also equals the fk colset.  Try to stick to one set-manipulation paradigm within one routine like this...